### DISCUSSION TITLE
Single Epoch Output Not Permitted

### DISCUSSION THREAD

**Discussion by njlai** [galacticusorg/galacticus#634] [Q&A] [answered]
Hello, sorry I am new to this. What would an error that reads "Single Epoch Output Not Permitted" refer to? This occurs when using the "outputAnalysis value="massFunctionStellarBaldry2012GAMA"

Thanks for the help.

**Comment by njlai** [comment]
Nvm i got it

**Reply by abensonca** [reply, marked as answer]
For the benefit of anyone else finding this thread, the `massFunctionStellarBaldry2012GAMA` class (and many other similar classes) require that there be 2 or more output times in the redshift range that was sampled by the observational study. (For `massFunctionStellarBaldry2012GAMA` that would be $z=0$ to $z=0.06$.) This is to allow the analysis to capture the evolution in the mass function across the redshift range. 

I'll update the error message here to make it more helpful.

**Comment by abensonca** [comment]
I've updated the code with more informative error messages for these types of issue.
